js/main.js

```diff
@@ -0,0 +1,733 @@
+
+function errorFocusOutHandler() {
+  hideErrorPlaceholder.style.display = "inline-block";
+}
+
+errorInFocus.addEventListener("focusout", errorFocusOutHandler);
+
+
+
+// navigation
+
+var avatarSection = document.querySelector("#avatar-list");
+var alertSection = document.querySelector("#alert-list");
+var badgeSection = document.querySelector("#badge-list");
+var buttonSection = document.querySelector("#button-list");
+var cardSection = document.querySelector("#card-list");
+var imageSection = document.querySelector("#image-list");
+var inputSection = document.querySelector("#input-list");
+var listSection = document.querySelector("#lists-list");
+var navigationSection = document.querySelector("#navigation-list");
+var modalSection = document.querySelector("#modal-list");
+var ratingSection = document.querySelector("#rating-list");
+var sliderSection = document.querySelector("#slider-list");
+var gridSection = document.querySelector("#grid-list");
+var utilitiesSection = document.querySelector("#text-utilities-list");
+var toastSection = document.querySelector("#toast-list");
+
+avatarSection.classList.add("sidebar-list-active");
+
+function avatarSectionHandler() {
+  avatarSection.classList.add("sidebar-list-active");
+  alertSection.classList.remove("sidebar-list-active");
+  badgeSection.classList.remove("sidebar-list-active");
+  buttonSection.classList.remove("sidebar-list-active");
+  cardSection.classList.remove("sidebar-list-active");
+  imageSection.classList.remove("sidebar-list-active");
+  inputSection.classList.remove("sidebar-list-active");
+  listSection.classList.remove("sidebar-list-active");
+  navigationSection.classList.remove("sidebar-list-active");
+  modalSection.classList.remove("sidebar-list-active");
+  ratingSection.classList.remove("sidebar-list-active");
+  sliderSection.classList.remove("sidebar-list-active");
+  gridSection.classList.remove("sidebar-list-active");
+  utilitiesSection.classList.remove("sidebar-list-active");
+  toastSection.classList.remove("sidebar-list-active");
+
+  document.querySelector("#avatar").style.display = "block";
+  document.querySelector("#alert").style.display = "none";
+  document.querySelector("#badge").style.display = "none";
+  document.querySelector("#button").style.display = "none";
+  document.querySelector("#card").style.display = "none";
+  document.querySelector("#image").style.display = "none";
+  document.querySelector("#input").style.display = "none";
+  document.querySelector("#list").style.display = "none";
+  document.querySelector("#navigation").style.display = "none";
+  document.querySelector("#modal").style.display = "none";
+  document.querySelector("#rating").style.display = "none";
+  document.querySelector("#slider").style.display = "none";
+  document.querySelector("#grid").style.display = "none";
+  document.querySelector("#text-utilities").style.display = "none";
+  document.querySelector("#toast").style.display = "none";
```

Comment:
This can be refactored as:

```suggestion
const sidebarItems =  document.querySelectorAll( ".sidebar-nav-unordered-list .sidebar-nav-item" );
const mainContentSections =  document.querySelectorAll( ".main-content .section-container" );

// Toggle active class for list items
sidebarItems.forEach(listItem => listItem.classList.remove("sidebar-list-active") ); 
avatarSection.classList.add("sidebar-list-active");

// Toggle display for main content sections
mainContentSections.forEach(section => section.style.display = "none");
document.querySelector("#avatar").style.display = "block";
```

Similar refactoring can be done for the other component section handlers as well.